test/e2e/storage/persistent_volumes.go

```diff
@@ -318,9 +320,34 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
 
 	ginkgo.Describe("Default StorageClass", func() {
 		ginkgo.Context("pods that use multiple volumes", func() {
+			var (
+				defaultStorageClass *storagev1.StorageClass
+				// only set for windows
+				disposeWindowsStorageClass func()
+				windowsDefaultStorageClass *storagev1.StorageClass
+			)
+
+			ginkgo.BeforeEach(func() {
+				// assumes that there's a default storage class already installed in the cluster
+				// in GCE the default StorageClass defaults to using fsType=ext4
+				defaultStorageClass, _ = testsuites.SetupStorageClass(c, nil)
+				if framework.NodeOSDistroIs("windows") {
+					// for windows the fsType should default to ntfs:
+					// - create a new storage class and set fsType=ntfs
+					// - make the created storaged class the default storage class
+					windowsDefaultStorageClass = newStorageClass(testsuites.StorageClassTest{}, f.Namespace.Name, "sc-windows")
+					windowsDefaultStorageClass, disposeWindowsStorageClass = testsuites.SetupStorageClass(f.ClientSet, windowsDefaultStorageClass)
+					e2epv.SetAsDefaultStorageClass(f.ClientSet, windowsDefaultStorageClass)
```

Comment:
is this really necessary? I think starting from 1.20 or 1.21, default storage class for linux can also work for windows. No need to set fstype explicitly.

Reply:
the default StorageClass that the cluster has is:

```
 allowVolumeExpansion: true
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   annotations:
     storageclass.kubernetes.io/is-default-class: "true"
   creationTimestamp: "2021-03-17T17:41:27Z"
   labels:
     addonmanager.kubernetes.io/mode: EnsureExists
   name: standard
   resourceVersion: "5105323"
   uid: 9b6ca4e7-80b3-4164-b5a7-53ba4af085a1
 parameters:
   type: pd-standard
 provisioner: kubernetes.io/gce-pd
 reclaimPolicy: Delete
 volumeBindingMode: Immediate
```

It doesn't have the `fsType` parameter however when I run it in a windows node it still tries to use `ext4`

```
Warning  FailedMount             1s (x2 over 6s)    kubelet, e2e-test-mauriciopoppe-windows-node-group-b5wm  MountVolume.MountDevice failed for volume "pvc-96
 c14ef2-fe7a-470f-b5b9-e0a7dd6c0af1" : rpc error: code = Internal desc = Failed to format and mount device from ("4") to ("\\var\\lib\\kubelet\\plugins\\kubernet
 es.io\\csi\\pv\\pvc-96c14ef2-fe7a-470f-b5b9-e0a7dd6c0af1\\globalmount") with fstype ("ext4") and options ([]): GCE PD CSI driver can only supports ntfs file sys
 tem, it does not support ext4
```

Reply:
oh, right, I think in-tree driver still has this issue. CSI driver fixed it.